sycl/test-e2e/Experimental/launch_queries/test_plan.md

```diff
@@ -0,0 +1,72 @@
+# Test plan for [`sycl_ext_oneapi_launch_queries`][spec-link] extension
+
+## Testing scope
+
+### Device coverage
+
+The unit tests should be launched on every supported device configuration we
+have.
+
+### Type coverage
+
+Each query is templated on a single template type argument `Param`
+with some queries being template overloads of each other.
+Each query has a restriction on the value of `Param` defined
+in the relevant section of the spec. 
+
+Param must be one of the following types defined in 
+`sycl::ext::oneapi::experiemntal::info::kernel` namespace:
+- `max_work_item_sizes<1>`
+- `max_work_item_sizes<2>`
+- `max_work_item_sizes<3>`
+- `max_work_group_size`
+- `max_num_work_groups`
+- `max_sub_group_size`
+- `num_sub_groups`
+
+The tests should cover all of these types.
+
+## Tests
+
+### Unit tests
+
+#### Interface tests
+
+These tests are intended to check that all classes and methods defined by the
+extension have correct implementation, i.e.: right signatures, right return
+types, all necessary constraints are checked/enforced, etc.
+
+These tests should check the following:
+
+- that each query is not available when the template argument `Param` has
+  a value different than the one in the spec.
+- that each query can be called with the appropriate value for `Param` and the
+  appropriate argument types as defined by its signature.
+- the return types of all queries match the spec.
+
+Tests in this category may not perform some useful actions to exercise the
+extension functionality in full, but instead they are focused on making sure
+that all APIs are consistent with respect to other APIs.
+
+#### Consistency of `max_subgroup_size` queries
```

Comment:
I think that we should also have a test which checks consistency of those queries comparing to device-specific limits - queue-kernel specific limit cannot be higher than device-specific limit (for a device associated with a queue)